File: Practice/Stack/DoubleCharacterTrouble.js
let DoubleCharacterTrouble = function (A) {
  let stack = [];
  for (const c of A) {
    if (stack.length == 0) stack.push(c);
    else {
      let top = stack[stack.length - 1];
      if (top == c) {
        stack.pop();
      } else stack.push(c);
    }
  }

  return stack.join("");
};

let str = `abbabbcd`;
console.log(DoubleCharacterTrouble(str));

File: Practice/Stack/DoubleCharacterTrouble.java
import java.util.Stack;

public class DoubleCharacterTrouble {
    public String solve(String A) {
        Stack<Character> stack = new Stack<>();

        for (int i = 0; i < A.length(); i++) {
            char currentValue = A.charAt(i);
            if (!stack.isEmpty() && stack.peek() == currentValue) {
                stack.pop();
            } else {
                stack.push(currentValue);
            }
        }
        StringBuilder output = new StringBuilder();
        while (!stack.isEmpty()) {
            output.insert(0, stack.pop());
        }

        return output.toString();
    }
}